dev/run-tests.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import re
+import sys
+import shutil
+import subprocess
+from collections import namedtuple
+
+SPARK_PROJ_ROOT = \
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+USER_HOME_DIR = os.environ.get("HOME")
+
+SBT_MAVEN_PROFILE_ARGS_ENV = "SBT_MAVEN_PROFILES_ARGS"
+AMPLAB_JENKINS_BUILD_TOOL = os.environ.get("AMPLAB_JENKINS_BUILD_TOOL")
+AMPLAB_JENKINS = os.environ.get("AMPLAB_JENKINS")
+
+SBT_OUTPUT_FILTER = re.compile("^.*[info].*Resolving" + "|" + 
+                               "^.*[warn].*Merging" + "|" +
+                               "^.*[info].*Including")
+
+
+def get_error_codes(err_code_file):
+    """Function to retrieve all block numbers from the `run-tests-codes.sh`
+    file to maintain backwards compatibility with the `run-tests-jenkins` 
+    script"""
+    
+    with open(err_code_file, 'r') as f:
+        err_codes = [e.split()[1].strip().split('=') 
+                     for e in f if e.startswith("readonly")]
+        return dict(err_codes)
+
+
+def exit_from_command_with_retcode(cmd, retcode):
+    print "[error] running", cmd, "; received return code", retcode
```

Comment:
From @nchammas above:

> Python 2.6 is the oldest version of Python that Spark officially supports.
> 
> We also added Python 3 support recently, so ideally this script should be able to run on 2.6+ and 3.3+, but I think it's fine to start with just 2.6+ since this is a developer script.

Given that maybe we should explicitly call out `python2` at the top?